### PR TITLE
Update docs to take into account CO:RE default

### PR DESCRIPTION
**Pull request by grantseltzer**
Signed-off-by: grantseltzer <grantseltzer@gmail.com>

### CODE REVIEW

File: docs/install/ebpf-compilation.md [hunk 1]
```diff
@@ -21,11 +24,5 @@ In this case, the full Docker image can be replaced by the lighter-weight `aquas
 If using Docker, the following `docker run` options demonstrate mounting a pre-compiled eBPF artifact into the container, and pointing Tracee to use it:
 
 ```bash
-docker run ... -v /path/in/host/tracee.bpf.123.o:/path/in/container/tracee.bpf.o -e TRACEE_BPF_FILE=/path/in/container/tracee.bpf.o aquasec/tracee
+docker run ... -v /path/in/host/tracee.bpf.123.o:/path/in/container/tracee.bpf.o aquasec/tracee
```
Comment (yanivagman) [inline docs/install/ebpf-compilation.md:27]: Why did you remove the environment variable?

**Review by yanivagman — COMMENTED** [review]
Don't we also want to update the one liner (both in the readme and docs) to start tracee? If a kernel has BTF enabled, it is possible to remove the linux headers mounts

File: docs/install/ebpf-compilation.md (hunk 1 above)
Comment (grantseltzer) [reply docs/install/ebpf-compilation.md:27]: The CO:RE object is embedded, so it's not needed to run

Comment (grantseltzer) [reply docs/install/ebpf-compilation.md:27]: On second thought, since this section is for mounting the precompiled binary, it should stay. Good call out.

File: Readme.md [hunk 2]
```diff
@@ -22,10 +22,18 @@ The full documentation of Tracee is available at [https://aquasecurity.github.io
...
+If running on BTF enabled kernel:
+
 ```bash
-docker run --name tracee --rm --privileged -v /lib/modules/:/lib/modules/:ro -v /usr/src:/usr/src:ro -v /tmp/tracee:/tmp/tracee -it aquasec/tracee:latest
+docker run --name tracee --rm --privileged -it aquasec/tracee:latest
 ```
 
+If running on BTF disabled kernel:
+
+```
```
Comment (yanivagman) [inline Readme.md:33]: Missing 'bash'?

**Review by yanivagman — APPROVED** [review]
LGTM!

File: docs/install/prerequisites.md [hunk 3]
```diff
@@ -1,11 +1,18 @@
 # Prerequisites
 
+Portable (CO:RE) option:
 - Linux kernel version >= 4.18
-- Linux kernel headers available under conventional location (see [Linux Headers](../headers) section for more info)
+- `CONFIG_DEBUG_INFO_BTF` Linux compile option enabled (only needed for running, not compilation)
```
Comment (itaysk) [inline docs/install/prerequisites.md:5]: IMO the kernel config is not the requisite, BTF is. I suggest something similar to my other comment about this

File: docs/install/prerequisites.md [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
 - libc, and the libraries: libelf, zlib
 - GNU Make >= 4.3
-- clang >= 9
+- clang >= 11
```
Comment (itaysk) [inline docs/install/prerequisites.md:8]: I think with CO:RE we can start to remove the compile-time requirements from here, since this is the first thngs the user sees about tracee, and it's quite intimidating. wdyt about moving this to the ebpf-compilation doc?

File: Readme.md [hunk 5]
```diff
@@ -22,10 +22,18 @@ The full documentation of Tracee is available at [https://aquasecurity.github.io
...
-docker run --name tracee --rm --privileged -v /lib/modules/:/lib/modules/:ro -v /usr/src:/usr/src:ro -v /tmp/tracee:/tmp/tracee -it aquasec/tracee:latest
+docker run --name tracee --rm --privileged -it aquasec/tracee:latest
 ```
 
+If running on BTF disabled kernel:
+
+```bash
+docker run --name tracee --rm --privileged -v /lib/modules/:/lib/modules/:ro -v /usr/src:/usr/src:ro -v /tmp/tracee:/tmp/tracee -it aquasec/tracee:latest
+
+```
```
Comment (itaysk) [inline Readme.md:36]: IMO remove from the quickstart, and link to the doc If non-btf

**Comment by grantseltzer** [comment]
Noting here that it appears libbpf needs access to the Kconfig (`/boot/config-$(uname -r)`) for CO:RE to work.

**Comment by itaysk** [comment]
as a general note, all of the comments are just because I'm considering these docs user facing and not developer (contributor) facing.

File: docs/tracee-ebpf/index.md [hunk 6]
```diff
@@ -9,7 +9,7 @@ In some cases, you might want to leverage Tracee's eBPF event collection capabil
 Before you proceed, make sure you follow the [minimum requirements for running Tracee](install/prerequisites.md).
 
 ```bash
-docker run --name tracee --rm --privileged -v /lib/modules/:/lib/modules/:ro -v /usr/src:/usr/src:ro -v /tmp/tracee:/tmp/tracee -it aquasec/tracee:latest trace
+docker run --name tracee --rm --privileged -it aquasec/tracee:latest trace
```
[the remark this answers is not in the document]
Comment (grantseltzer) [reply docs/tracee-ebpf/index.md:12]: It seems that sysfs is automatically read only mounted.

File: Readme.md (hunk 5 above)
Comment (grantseltzer) [reply Readme.md:36]: Which doc do you think should be linked to?

Comment (grantseltzer) [reply Readme.md:36]: Perhaps a new document that talks about the differences between CO:RE and kernel-header-dependent versions of tracee
